### PR TITLE
Add testing for redux reducers and store

### PR DESCRIPTION
**Pull request by dlwub**
Quizzical enables users:
- To fetch questions from the Open Trivia database based on the category they choose
- To select answers from the given choices
- To see the correct and incorrect choices they made
- To return to the initial page 
Further:
* A test file is added
* The project is deployed and the link is attached to the README file
* A video link describing the project is attached to the README file

### CODE REVIEW

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @dlwub  👏🏽, 
## Good job so far! 👍🏽 
### Summary ♨️ 
- No Linter error ✔️ 
- Good and descriptive PR 💯 
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## What can be improved on
- Kindly make sure you merge all your branches to the development branch then ask for a review from there. This will enable easier and inline code review 😊
- Kindly make sure all your test are passing before asking for another review 😉



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codecaiine — COMMENTED** [review]
Hi @dlwub , 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
#### Please apply my comment carefully :smile: 
- [ ] Please make sure that the project files are well organized and give them names that can help someone who is new in your code pages like homes and details page :smile: 

- [ ] From the project requirement you need to have on your home page when the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name. Please fix it :+1: 
- [ ] From the project requirement in your details page when the user clicks on the "Back" button (<), the user navigates to the home page. Please fix  it :smile: 

_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/components/Question.js [hunk 1]
```diff
@@ -34,4 +34,5 @@ Question.propTypes = {
   id: PropTypes.string.isRequired,
   question: PropTypes.string.isRequired,
   choices: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
+  // correctAnswer: PropTypes.string.isRequired,
```
[the remark this answers is not in the document]
Comment (dlwub) [reply src/components/Question.js:37]: Hi @ codecaiine,
Thank you for the helpful feedback you gave me. While most of your comments are beneficial to improve my work, I disagree with some of the points that you raised:
- I agree that I had to remove commented codes and the test file was missing (I already included that now)
- However, the file is well organized (regardless of the file names)
- The app shows the home page when it loads and the home page has a list of categories (General knowledge, Entertainment, Science, and Social). The data is filtered based on the button that you click on. The questions one gets by clicking the 'General Knowledge' button are different from the questions that one gets when clicking the 'Science' button.
- A 'Play Again' button appears on the questions page after the user chooses answers and clicks on the 'Check Answer' button. It returns the user to the home page (it replaces the back button)
- The buttons on the home page are used as filters (as I mentioned above)
- I think the design I used is similar to the template design. Similar doesn't mean identical or the same. I agree that I don't have maps of countries on my home page. If we have to use the same template, we had not to be given the freedom to choose our own theme. I chose my own theme and I included pictures that can represent my theme. Also, I chose my own color set and fonts. 
Thank you once more again for your thorough comments.

[the remark this answers is not in the document]
Comment (codecaiine) [reply src/components/Question.js:37]: Hi, @dlwub Thank you for your comments. 
- First, it was difficult for me to navigate into your project. I can not. Your structure was not clear enough for me. 
- Secondly From the project requirement you need to have on your home page when the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name you did well but you missed adding a filter like the one on the screenshots below which is part of project requirement ...
194139630-038c25cc-bcf7-429c-8978-244f86bf103e

**Review by Omar-Muhamad — COMMENTED** [review]
# Invalid ❌

Hi @, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

Your Code Review Request is marked as invalid for these reasons:

- [ ] Gitflow is not correctly used. Please make this Pull Request from the dev branch to the main branch and be sure that it contains every change you did to the project.


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

**Pull request merged**